File: db/csharp/SalesManagement.Domain/Models/AlternateProduct.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 代替商品のEntityクラス
    /// </summary>
    public class AlternateProduct
    {
        public string ProductCode { get; set; } = string.Empty;
        public string AlternateProductCode { get; set; } = string.Empty;
        public int Priority { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: db/fsharp/SalesManagement.Domain/Models/AlternateProduct.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 代替商品
/// </summary>
[<CLIMutable>]
type AlternateProduct = {
    ProductCode: string
    AlternateProductCode: string
    Priority: int
    CreatedAt: DateTime
    CreatedBy: string
    UpdatedAt: DateTime
    UpdatedBy: string
}
